streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -235,12 +242,35 @@ public String toString() {
     }
 
     /**
+     * Cleans up local state store directory ({@code state.dir}), by deleting all data with regard to the application-id.
+     * <p>
+     * May only be called either before instance is started or after instance is closed.
+     *
+     * @throws IllegalStateException if instance is currently running
+     */
+    public void cleanUp() {
```

Comment:
If it only cleans the local state directory, do we want to rename it to sth like `cleanLocalState()`? (just asking)